### PR TITLE
add new command label

### PR DESCRIPTION
**Pull request by zhzhuang-zju**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
add new command label

**Which issue(s) this PR fixes**:
Parts of  #5249

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
`karmadactl`: add a new command `label` to update the labels on a resource
```

### CODE REVIEW

**Comment by zhzhuang-zju** [comment]
assign @RainbowMango

File: pkg/karmadactl/label/label.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	kubectllabel "k8s.io/kubectl/pkg/cmd/label"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	"github.com/karmada-io/karmada/pkg/karmadactl/util"
+)
+
+var (
+	labelExample = templates.Examples(`
+		# Update propagationpolicy 'foo' with the label 'priority' and the value '0'
+		[1]%s label pp foo priority=0
```
Comment (RainbowMango) [inline pkg/karmadactl/label/label.go:30]: I'd say this seems not a good example, as this label might be misleading with the priority in PropagationPolicy.

Actually, we have some real-world labels users might use, like `resourcetemplate.karmada.io/deletion-protected`.

File: pkg/karmadactl/label/label.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		# Update deployment 'foo' only if the resource is unchanged from version 1
+		[1]%s label deployment resourcetemplate.karmada.io/deletion-protected=Always --resource-version=1
+
+		# Update deployment 'foo' by removing a label named 'bar' if it exists
+		# Does not require the --overwrite flag
+		[1]%s label deployment foo resourcetemplate.karmada.io/deletion-protected-`)
+)
+
+// NewCmdLabel returns new initialized instance of label sub command
+func NewCmdLabel(f util.Factory, parentCommnd string, ioStreams genericiooptions.IOStreams) *cobra.Command {
```
Comment (RainbowMango) [inline pkg/karmadactl/label/label.go:50]:
```suggestion
func NewCmdLabel(f util.Factory, parentCommand string, ioStreams genericiooptions.IOStreams) *cobra.Command {
```
Typo, just get it from #5454

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**